src/api/logs.rs

```diff
@@ -0,0 +1,329 @@
+        cfg: Arc<RwLock<Config>>,
+        pk: String,
+    ) -> Result<Log, Response<Body>> {
+        let payload: String = String::from_utf8(entire_body.to_vec())
+            .map_err(|_| return_400("Could not understand request"))?;
+        let read_cfg = cfg.read().unwrap();
+        let mut current_log = match read_cfg.log.get(&pk) {
+            Some(v) => v.clone(),
+            None => {
+                return Err(return_404());
+            }
+        };
+
+        let log: serde_json::Value =
+            serde_json::from_str(&payload).map_err(|_| return_400("Could not parse request"))?;
+
+        // Commit Window
+        if let Some(serde_json::Value::String(commit_window)) = log.get("commit_window") {
+            // Validate Commit Window
+            if commit_window == "" {
+                return Err(return_400("Commit window key cannot be empty."));
+            }
+            if !commit_window.ends_with("s") && !commit_window.ends_with("m") {
+                return Err(return_400(
+                    "Commit window must be specified in either seconds `5s` or minutes `1m`",
+                ));
+            }
+            // if the commit window parses to 0 and the value is not 0, 0s or 0m, it's an invalid window
+            let parsed_window = Config::commit_window_to_seconds(&commit_window);
+            if commit_window != "0"
+                && commit_window != "0s"
+                && commit_window != "0m"
+                && parsed_window == 0
+            {
+                return Err(return_400("Commit window is invalid"));
+            }
+            current_log.commit_window = commit_window.clone();
+        }
+
+        let cfg_read = cfg.read().unwrap();
+        // validate the datastores
+        if let Some(serde_json::Value::Array(datastores_value)) = log.get("datastores") {
+            for ds_name_value in datastores_value {
+                if let serde_json::Value::String(ds_name) = ds_name_value {
+                    if cfg_read.datastore.contains_key(ds_name) == false {
+                        return Err(return_400(&format!(
+                            "{} is an invalid datastore name",
+                            &ds_name
+                        )));
+                    }
+                }
+            }
+        }
+
+        // Validate name
+        let mut log_name: Option<String> = None;
+        if let Some(serde_json::Value::String(name)) = log.get("name") {
+            if name == "" {
+                return Err(return_400("Log name cannot be empty."));
+            }
+            current_log.name = Some(name.clone());
+            log_name = Some(name.clone());
+        }
+
+        // if log name changed, delete previous file
+        if let Some(ds_name) = log_name {
+            if ds_name != pk {
+                let cfg = Arc::clone(&cfg);
+                tokio::spawn({
+                    delete_object_metabucket(cfg, format!("minsql/meta/logs/{}", pk))
```

Comment:
Since we dont wait for this, I think this may complete after the update in the handler causing a race.

Reply:
We don't need to wait for it, once the delete happens, the `listen_bucket_notification` will notify `meta` to update it's contents